Assets/Scripts/Player/PlayerController.cs

```diff
@@ -111,7 +120,54 @@ private void Update()
             Vector3 playerPosition = transform.position;
             _camera.transform.position = playerPosition + _cameraPosition;
             _camera.transform.LookAt(playerPosition);
+
+            // Check and makes character facing the target object when moving toward it
+            if (target != null) FaceTarget(target.gameObject, true, out _);
+        }
+
+        /// <summary>
+        /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+        /// (Won't conflict with character attacking enemy or moving on the map)
+        /// </summary>
+        #region Interactables Functions
+        
+        // This will make character going to target object and stop when it is close enough
+        public void SetTarget(Interactable newTarget)
+        {
+            _navMeshAgent.stoppingDistance = newTarget.Radius * .8f;
+            _navMeshAgent.updateRotation = false;
+            target = newTarget.transform;
+        }
+
+        // Makes character not targeting the object anymore
+        public void StopTarget()
+        {
+            _navMeshAgent.stoppingDistance = 0f;
```

Comment:
you set the stoppingDistance to 0 😮 ?

Reply:
This basically mean that if the character not targeting anything then it will be switched back to normal, that means the character will not stop before reaching the position where the mouse clicked.

Reply:
You're right, default stoppingDistance is 0. I confused things here!